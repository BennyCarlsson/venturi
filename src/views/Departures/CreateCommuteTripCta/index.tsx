import Colors from 'tokens/colors'
import {
  CreateCommuteTripCtaWrapper,
  StyledParagraph,
  StyledPlusIcon,
} from './styles'

type CreateCommuteTripCtaProps = {
  onClick: () => void
}

const CreateCommuteTripCta = ({ onClick }: CreateCommuteTripCtaProps) => {
  return (
    <CreateCommuteTripCtaWrapper
      onClick={(e) => {
        e.preventDefault()
        onClick()
      }}
    >
      <StyledPlusIcon />
      <StyledParagraph fontSize={18} color={Colors.white}>
        Create commute trip
      </StyledParagraph>
    </CreateCommuteTripCtaWrapper>
  )
}

export default CreateCommuteTripCta
