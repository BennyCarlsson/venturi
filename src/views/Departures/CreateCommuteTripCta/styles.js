import Paragraph from 'components/Paragraph'
import styled from 'styled-components'
import Colors from 'tokens/colors'
import spacings from 'tokens/spacings'
import { ReactComponent as PlusIcon } from 'assets/icons/plus.svg'

export const CreateCommuteTripCtaWrapper = styled.a`
  position: absolute;
  bottom: ${spacings.bottomMenuHeight};
  left: 0;
  width: 100%;
  display: flex;
  align-items: center;
  background-color: ${Colors.vBlack};
  border-bottom: 1px solid white;
`

export const StyledPlusIcon = styled(PlusIcon)`
  stroke: white;
  margin-left: 40px;
`

export const StyledParagraph = styled(Paragraph)`
  padding: 14px 21px;
`
