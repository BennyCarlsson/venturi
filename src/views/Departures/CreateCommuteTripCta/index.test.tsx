import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import CreateCommuteTripCta from ".";

describe("<CreateCommuteTripCta />", () => {
  const onClick = jest.fn();
  it("should match snapshot", () => {
    const tree = renderer
      .create(<CreateCommuteTripCta onClick={onClick} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render without crashing", () => {
    render(<CreateCommuteTripCta onClick={onClick} />);
    expect(screen).toBeTruthy();
  });

  it("should call onClick onClick", () => {
    expect(onClick).not.toHaveBeenCalled();
    render(<CreateCommuteTripCta onClick={onClick} />);
    userEvent.click(screen.getByText("Create commute trip"));
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
