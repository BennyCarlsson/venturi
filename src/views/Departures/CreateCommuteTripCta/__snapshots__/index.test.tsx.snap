// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateCommuteTripCta /> should match snapshot 1`] = `
<a
  className="sc-gsTCUz jxAhtz"
  onClick={[Function]}
>
  <svg
    className="sc-dlfnbm ilDCCC"
  >
    plus.svg
  </svg>
  <p
    className="sc-bdfBwQ gbRIVK sc-hKgILt SVDlr"
    color="#FFFFFF"
    fontSize={18}
  >
    Create commute trip
  </p>
</a>
`;
